src/librustc_passes/entry.rs

```diff
@@ -48,7 +48,7 @@ impl<'a, 'tcx> ItemLikeVisitor<'tcx> for EntryContext<'a, 'tcx> {
     }
 }
 
-fn entry_fn(tcx: TyCtxt<'_>, cnum: CrateNum) -> Option<(DefId, EntryFnType)> {
+fn entry_fn(tcx: TyCtxt<'_>, cnum: CrateNum) -> Option<(LocalDefId, EntryFnType)> {
     assert_eq!(cnum, LOCAL_CRATE);
```

Comment:
Oh lol, okay this explains why the return type can be `LocalDefId`. Maybe the parameter should be removed, if there is only one possible value anyway?

Reply:
Yes ultimately we should remove the the param, but i haven't tried yet. Allegedly ,the query macros do not handle "no key at all" in a query

Reply:
I'll try to look at it. :+1:

Reply:
> Allegedly ,the query macros do not handle "no key at all" in a query

As a work-around, the key could be of type `()`. ;)

Reply:
@RalfJung query keys of type `()` are new, at the start the system only supported `DefId` and `CrateNum`, so queries that always take `LOCAL_CRATE` are an artifact of that.

Reply:
We can also make the query system variadic by auto-tupling behind the scenes and hiding the tupling from the `tcx` methods and the providers :D.